examples/clk/build.zig

```diff
@@ -0,0 +1,147 @@
+//
+// Copyright 2024, UNSW
+// SPDX-License-Identifier: BSD-2-Clause
+//
+const std = @import("std");
+
+const MicrokitBoard = enum { odroidc4, maaxboard };
+
+const Target = struct {
+    board: MicrokitBoard,
+    zig_target: std.Target.Query,
+};
+
+const targets = [_]Target{
+    .{
+        .board = MicrokitBoard.odroidc4,
+        .zig_target = std.Target.Query{
+            .cpu_arch = .aarch64,
+            .cpu_model = .{ .explicit = &std.Target.arm.cpu.cortex_a55 },
+            .os_tag = .freestanding,
+            .abi = .none,
+        },
+    },
+    .{
+        .board = MicrokitBoard.maaxboard,
+        .zig_target = std.Target.Query{
+            .cpu_arch = .aarch64,
+            .cpu_model = .{ .explicit = &std.Target.arm.cpu.cortex_a53 },
+            .os_tag = .freestanding,
+            .abi = .none,
+        },
+    },
+};
+
+fn findTarget(board: MicrokitBoard) std.Target.Query {
+    for (targets) |target| {
+        if (board == target.board) {
+            return target.zig_target;
+        }
+    }
+
+    std.log.err("Board '{}' is not supported\n", .{board});
+    std.posix.exit(1);
+}
+
+const ConfigOptions = enum { debug, release, benchmark };
+
+pub fn build(b: *std.Build) void {
+    const optimize = b.standardOptimizeOption(.{});
+
+    // Getting the path to the Microkit SDK before doing anything else
+    const microkit_sdk_arg = b.option([]const u8, "sdk", "Path to Microkit SDK");
+    if (microkit_sdk_arg == null) {
+        std.log.err("Missing -Dsdk=/path/to/sdk argument being passed\n", .{});
+        std.posix.exit(1);
+    }
+    const microkit_sdk = microkit_sdk_arg.?;
+
+    const microkit_config_option = b.option(ConfigOptions, "config", "Microkit config to build for") orelse ConfigOptions.debug;
+    const microkit_config = @tagName(microkit_config_option);
+
+    // Get the Microkit SDK board we want to target
+    const microkit_board_option = b.option(MicrokitBoard, "board", "Microkit board to target");
+
+    if (microkit_board_option == null) {
+        std.log.err("Missing -Dboard=<BOARD> argument being passed\n", .{});
+        std.posix.exit(1);
+    }
+    const target = b.resolveTargetQuery(findTarget(microkit_board_option.?));
+    const microkit_board = @tagName(microkit_board_option.?);
+
+    const microkit_board_dir = b.fmt("{s}/board/{s}/{s}", .{ microkit_sdk, microkit_board, microkit_config });
+    const microkit_tool = b.fmt("{s}/bin/microkit", .{microkit_sdk});
+    const libmicrokit = b.fmt("{s}/lib/libmicrokit.a", .{microkit_board_dir});
+    const libmicrokit_include = b.fmt("{s}/include", .{microkit_board_dir});
+    const libmicrokit_linker_script = b.fmt("{s}/lib/microkit.ld", .{microkit_board_dir});
+
+    const dtb_arg = b.option([]const u8, "dtb", "Path to DTB file");
+    if (dtb_arg == null) {
+        std.log.err("Missing -Dsdk=/path/to/dtb argument being passed\n", .{});
```

Comment:
Typo here - this should be `Missing -Ddtb` not `Missing -Dsdk`. This stuff should also be in a README.md for this example but there isn't one